scipy/io/tests/test_mmio.py

```diff
@@ -387,6 +387,20 @@ def test_sparse_formats(self):
                 result = mmread(fn).todense()
                 assert_array_almost_equal(result, expected)
 
+    def test_precision(self):
+        for precision in range(1, 8):
+            # construct sparse matrix with pi as last main diagonal entry
+            n = 10**precision + 1
+            A = scipy.sparse.lil_matrix((n, n))
+            A[n-1, n-1] = pi
+            # write matrix with low precision and read again
+            mmwrite(self.fn, A, precision=precision)
+            A = scipy.io.mmread(self.fn)
+            # check for right entries in matrix
+            assert_array_equal(A.row, [n-1])
+            assert_array_equal(A.col, [n-1])
+            assert_array_almost_equal(A.data, [pi], decimal=precision-1)
```

Comment:
You mean your test `assert_array_almost_equal(A.data, [np.round(np.pi, precision - 1)]`? This test will work for pi but not for every value. As example, try to run the following. For 0.01 you will get an error.
```
import numpy as np
v = np.pi
for precision in range(1,10):
    np.testing.assert_array_almost_equal([float('%%.%dg' % precision % v)], [np.round(v, precision - 1)])
v = 0.01
for precision in range(1,10):
    np.testing.assert_array_almost_equal([float('%%.%dg' % precision % v)], [np.round(v, precision - 1)])
```
```
Traceback (most recent call last):
  File "test.py", line 7, in <module>
    np.testing.assert_array_almost_equal([float('%%.%dg' % precision % v)], [np.round(v, precision - 1)])
  File "/usr/lib/python3/dist-packages/numpy/testing/utils.py", line 811, in assert_array_almost_equal
    header=('Arrays are not almost equal to %d decimals' % decimal))
  File "/usr/lib/python3/dist-packages/numpy/testing/utils.py", line 644, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Arrays are not almost equal to 6 decimals

(mismatch 100.0%)
 x: array([ 0.01])
 y: array([ 0.])
```